napari_builtins/_tests/test_io.py

```diff
@@ -104,24 +103,23 @@ def test_zarr_with_unrelated_file(tmp_path):
     np.testing.assert_array_equal(image, image_in[0])
 
 
-def test_zarr_multiscale():
+def test_zarr_multiscale(tmp_path):
     multiscale = [
         np.random.random((20, 20)),
         np.random.random((10, 10)),
         np.random.random((5, 5)),
     ]
-    with TemporaryDirectory(suffix='.zarr') as fout:
-        root = zarr.open_group(fout, 'a')
-        for i in range(len(multiscale)):
-            shape = 20 // 2**i
-            z = root.create_dataset(str(i), shape=(shape,) * 2)
-            z[:] = multiscale[i]
-        multiscale_in = magic_imread([fout])
-        assert len(multiscale) == len(multiscale_in)
-        # Note: due to lazy loading, the next line needs to happen within
-        # the context manager. Alternatively, we could convert to NumPy here.
-        for images, images_in in zip(multiscale, multiscale_in):
-            np.testing.assert_array_equal(images, images_in)
+    fout = str(tmp_path / 'multiscale.zarr')
+
+    root = zarr.open_group(fout, mode='a')
+    for i in range(len(multiscale)):
+        shape = 20 // 2**i
+        z = root.create_dataset(str(i), shape=(shape,) * 2)
```

Comment:
```suggestion
        z = root.create_array(str(i), shape=(shape,) * 2)
```
See: https://github.com/napari/napari/pull/7215/files#r1810721505

Reply:
I guess we can use `.create` like @Czaki did in the test_1d functions?

Reply:
I don't know. There is a lack of documentation. 
The `create` is also described as backward compatibility in v3.0 so still may emit some warning?